tutorials/sphinx-tutorials/multiagent_ppo.py

```diff
@@ -0,0 +1,789 @@
+#
+# For example, in this case, we will instantiate a ``RewardSum`` transform which will sum rewards over the episode.
+# We will tell this transform where to find the reward key and where to write the summed episode reward.
+# The transformed env will inherit
+# the device and meta-data of the wrapped env, and transform these depending on the sequence
+# of transforms it contains.
+#
+
+
+env = TransformedEnv(
+    env,
+    RewardSum(in_keys=[env.reward_key], out_keys=[("agents", "episode_reward")]),
+)
+
+
+######################################################################
+# the :func:`check_env_specs` function runs a small rollout and compares its output against the environment
+# specs. If no error is raised, we can be confident that the specs are properly defined:
+#
+check_env_specs(env)
+
+######################################################################
+# Rollout
+# ~~~~~~~
+#
+# For fun, let's see what a simple random rollout looks like. You can
+# call `env.rollout(n_steps)` and get an overview of what the environment inputs
+# and outputs look like. Actions will automatically be drawn at random from the action spec
+# domain.
+#
+n_rollout_steps = 5
+rollout = env.rollout(n_rollout_steps)
+print("rollout of three steps:", rollout)
+print("Shape of the rollout TensorDict:", rollout.batch_size)
+######################################################################
+# We can see that our rollout has ``batch_size`` of ``(num_vmas_envs, n_rollout_steps)``.
+# This means that all the tensors in it will have those leading dimensions.
+#
+# Looking more in depth, we can see that the output tensordict can be divided in the following way:
+#
+# - *In the root* (accessible by running ``rollout.exclude("next")`` ) we will find all the keys that are available
+#   after a reset is called at the first timestep. We can see their evolution through the rollout steps by indexing
+#   the ``n_rollout_steps`` dimension. Among these keys, we will find the ones that are different for each agent
+#   in the ``rollout["agents"]`` tensordict, which will have batch size ``(num_vmas_envs, n_rollout_steps, n_agents)``
+#   signifying that it is storing the additional agent dimension. The ones outside this agent tensordict
+#   will be the shared ones (in this case only done).
+# - *In the next* (accessible by running ``rollout.get("next")`` ). We will find the same structure as the root,
+#   but for keys that are available only after a step.
+#
+# In torchrl the convention is that done and observations will be present in both root and next (as these are
+# available both at reset time and after a step). Action will only be available in root (as there is no action
+# resulting from a step) and reward will only be available in next (as there is no reward at reset time).
+# This structure follows the one in the Sutton and Barto book where root represents data at time :math:`t` and
+# next represents data at time :math:`t+1` of a world step.
```

Comment:
Do we need to justify?
If so
```suggestion
# This structure follows the one in **Reinforcement Learning: An Introduction (Sutton and Barto)** where root represents data at time :math:`t` and
# next represents data at time :math:`t+1` of a world step.
```

Reply:
i thought it was cool to point it out to make users understand better. Cause i frequently get asked how things are separated